<launch>
  <!-- Arguments -->
  <!-- If use remote control, you should unable 'use_local_joy', 'open_rviz', 'open_reconfigure',  -->
  <!-- and run these nodes on remote computer -->
  <arg name="use_local_joy" default="true"/> <!--接收头插在小车工控机上-true；否则false-->
  <arg name="open_rviz" default="false"/> <!---->
  <arg name="open_reconfigure" default="false"/><!---->

  <arg name="use_local_controller" default="true"/><!--不想让底盘运动，或者使用tcp时-false-->
  <arg name="map_file" default="$(find agv_robot)/maps/map.yaml"/>

  <!-- joy control -->
  <node if="$(arg use_local_joy)" pkg = "agv_robot" type="joy_driver" name="joy_driver" output="screen"/>
  <!-- rviz -->
  <node if="$(arg open_rviz)" pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find agv_robot)/rviz/navigation.rviz"/>
  <!-- rqt_reconfigure -->
  <node if="$(arg open_reconfigure)" name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" />
  <!-- Serial node -->
  <node if="$(arg use_local_controller)" pkg = "agv_robot" type="base_control_node" name="base_control_node" output="screen">
    <param name="baudrate" type="int" value="115200" />
    <param name="port" type="string" value="/dev/ttyS0" />
  </node>

  <!-- tf -->
  <param name="robot_description" textfile="$(find agv_robot)/urdf/model.urdf"/>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <!-- main node -->
  <node pkg = "agv_robot" type="main" name="main" output="screen">
    <param name="base_foot_print" type="string" value="base_link" />
    <param name="odom_frame" type="string" value="odom" />
    <param name="map_frame" type="string" value="map" />
    <param name="publish_tf" type="bool" value="false" />
    <param name="max_joy_linear_velocity" type="double" value="0.4" />
    <param name="max_joy_angular_velocity" type="double" value="0.5" />
  </node>

  <!-- tcp client node -->
  <node pkg = "agv_robot" type="tcp_client_node" name="tcp_client_node" output="screen">
    <param name="server_ip" type="string" value="192.168.1.112" />
    <param name="port" type="int" value="9054" />
    <param name="cycle_ms" type="double" value="200" />
    <param name="map_name" type="string" value="$(arg map_file)"/>
  </node>
  <!-- Laserscan -->
  <include file="$(find agv_robot)/launch/laserscan.launch"/>
  <!-- imu -->
  <include file="$(find imu_interface_broadcaster)/launch/imu.launch"/>
  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>
  <!-- Cartographer localization -->
  <include file="$(find agv_robot)/launch/cartographer_localization.launch" />
  <!-- amcl localization -->
  <!--include file="$(find agv_robot)/launch/amcl.launch" /-->
  <!-- odom by scan_match -->
  <include file="$(find agv_robot)/launch/laser_scan_match.launch" />
  <!-- move_base -->
  <include file="$(find agv_robot)/launch/move_base.launch"/>
  <include file="$(find agv_robot)/launch/test.launch" />
</launch>
